pkg/sync/rpc/server.go

```diff
@@ -1503,7 +1511,13 @@ func (s *SyncAgentServer) SnapshotHashLockState(ctx context.Context, req *emptyp
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to try lock %v", fileLock)
 	}
-	defer fileLock.Unlock()
+	defer func() {
+		if unlockErr := fileLock.Unlock(); unlockErr != nil {
+			logrus.WithError(unlockErr).WithFields(logrus.Fields{
+				"file": fileLock.Path(),
+			}).Error("Failed to unlock file")
```

Comment:
```suggestion
			}).Warn("Failed to unlock file")
```

If we don't error out, using warning level message might be fine. WDYT?